lib/_http_client.js

```diff
@@ -889,12 +853,16 @@ function setSocketTimeout(sock, msecs) {
 }
 
 ClientRequest.prototype.setNoDelay = function setNoDelay(noDelay) {
-  this._deferToConnect('setNoDelay', [noDelay]);
+  this.once('socket', (socket) => {
```

Comment:
Won't this callback never get called if the socket is already signed when `setNoDelay()` is called? Ditto for `setSocketKeepAlive()`. There's also the problem if the socket is assigned but not connected yet.

Reply:
> Won't this callback never get called if the socket is already signed when 

You are right. We need a test for this.

> There's also the problem if the socket is assigned but not connected yet.

Why is that a problem?

Reply:
I vaguely recall one or both functions needing a connected socket before the settings can be applied? git blame might provide more answers.

Reply:
I'll dig into it.